vcfa/resource_vcfa_org_regional_networking_vpc_qos.go

```diff
@@ -0,0 +1,285 @@
+	vcfaClient := meta.(*VCDClient)
+	rns, err := vcfaClient.GetTmRegionalNetworkingSettingById(d.Id())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("error looking up %s by ID: %s", labelVcfaOrgNetworking, err)
+	}
+
+	// ID is Org Regional Networking Setting ID
+	d.SetId(rns.TmRegionalNetworkingSetting.ID)
+
+	// fetch VPC profile
+	vpcProfile, err := rns.GetDefaultVpcConnectivityProfile()
+	if err != nil {
+		return diag.Errorf("error retrieving %s: %s", labelVcfaOrgRegionalNetworkingVpcQos, err)
+	}
+
+	err = setTmOrgRegionalNetworkingVpcQosData(vcfaClient, d, vpcProfile)
+	if err != nil {
+		return diag.Errorf("error storing %s configuration to state: %s", labelVcfaOrgRegionalNetworkingVpcQos, err)
+	}
+
+	return nil
+}
+
+func resourceVcfaOrgRegionalNetworkingVpcQosDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcfaClient := meta.(*VCDClient)
+	rns, err := vcfaClient.GetTmRegionalNetworkingSettingById(d.Get("org_regional_networking_id").(string))
+	if err != nil {
+		return diag.Errorf("error looking up %s by ID: %s", labelVcfaOrgNetworking, err)
+	}
+
+	// ID is Org Regional Networking Setting ID
+	d.SetId(rns.TmRegionalNetworkingSetting.ID)
+
+	cfg, err := getTmOrgRegionalNetworkingVpcQosType(vcfaClient, rns, d, false)
+	if err != nil {
+		return diag.Errorf("error getting %s configuration: %s", labelVcfaOrgRegionalNetworkingVpcQos, err)
+	}
+
+	_, err = rns.UpdateDefaultVpcConnectivityProfile(cfg)
+	if err != nil {
+		return diag.Errorf("error removing custom %s configuration: %s", labelVcfaOrgRegionalNetworkingVpcQos, err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourceVcfaOrgRegionalNetworkingVpcQosImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	vcfaClient := meta.(*VCDClient)
+
+	id := strings.Split(d.Id(), ImportSeparator)
+	if len(id) != 2 {
+		return nil, fmt.Errorf("ID syntax should be <%s name>%s<%s name>", labelVcfaOrg, ImportSeparator, labelVcfaRegionalNetworkingSetting)
+	}
+
+	org, err := vcfaClient.GetTmOrgByName(id[0])
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving %s '%s': %s", labelVcfaOrg, id[0], err)
+	}
+
+	rns, err := vcfaClient.GetTmRegionalNetworkingSettingByNameAndOrgId(id[1], org.TmOrg.ID)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving %s '%s' within %s '%s': %s",
+			labelVcfaRegionalNetworkingSetting, id[1], labelVcfaOrg, id[0], err)
+	}
+
+	d.SetId(rns.TmRegionalNetworkingSetting.ID)
+	dSet(d, "org_regional_networking_id", rns.TmRegionalNetworkingSetting.ID)
+	return []*schema.ResourceData{d}, nil
+}
+
+// flag `isCreatedUpdate` is used to separate Create/Update and Delete operations
+func getTmOrgRegionalNetworkingVpcQosType(vcfaClient *VCDClient, rns *govcd.TmRegionalNetworkingSetting, d *schema.ResourceData, isCreatedUpdate bool) (*types.TmRegionalNetworkingVpcConnectivityProfile, error) {
```

Comment:
Just a dumb suggestion for readability
```suggestion
func getTmOrgRegionalNetworkingVpcQosType(vcfaClient *VCDClient, rns *govcd.TmRegionalNetworkingSetting, d *schema.ResourceData, isCreatedOrUpdated bool) (*types.TmRegionalNetworkingVpcConnectivityProfile, error) {
```

Reply:
Fixed.